R/zchunk_LB1233.Elec_water.R

```diff
@@ -40,76 +45,173 @@ module_gcam.usa_LB1233.Elec_water_DISABLED <- function(command, ...) {
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # adjust UCS database by resetting Michigan and Wisconsin power plant seawater use to surface water...
+    UCS_Database %>%
+      mutate(`Reported Water Source (Type)` = if_else(State %in% c("MI", "WI") &
+                                                        `Reported Water Source (Type)` == "Ocean",
+                                                      "Surface Water",
+                                                      `Reported Water Source (Type)`)) %>%
+      # make some further minor adjustments and introduce tech and water type names...
+      rename(state = State,
+             out_MWh = `Estimated Generation (MWh)`,
+             Technology.Code = `Technology Code`) %>%
+      right_join(UCS_tech_names, by = "Technology.Code") %>%
+      right_join(UCS_water_types, by = "Reported Water Source (Type)") %>%
+      # set water type to "none" for cases with no cooling system...
+      mutate(water_type = if_else(cooling_system == "none", "none", water_type),
+             # ...and change PV and hydro to fresh
+             water_type = if_else(technology %in% c("hydro", "PV"), "fresh", water_type)) ->
+      USC_db_adj
+
+    # aggregate and compute shares for states
+    USC_db_adj %>% group_by(state, sector, fuel, technology) %>%
+      summarise(out_MWh_ = sum(out_MWh)) %>% ungroup ->
+      L1233.out_MWh_state_elec_F_tech
+    USC_db_adj %>% group_by(state, sector, fuel, technology, cooling_system, water_type) %>%
+      summarise(out_MWh = sum(out_MWh)) %>% ungroup %>%
+      left_join_error_no_match(L1233.out_MWh_state_elec_F_tech,
+                               by = c("state", "sector", "fuel", "technology")) %>%
+      mutate(share = out_MWh / out_MWh_) %>% select(-out_MWh_) ->
+      L1233.out_MWh_state_elec_F_tech_cool
+
+    # aggregate and compute shares for FERC subregions, filtering for power plants built in the 2000s only
```

Comment:
why are we only interested in plants built in the 2000s?

Reply:
it might be worth defining FERC in the code once.

Reply:
not certain regarding post 2000s (no comment in the legacy code), but I'm pretty sure it's just so that the computed output values are contemporary. Comment added + FERC defined :+1: